apps/app/src/test/scala/org/lfdecentralizedtrust/splice/util/SvTestUtil.scala

```diff
@@ -191,6 +196,93 @@ trait SvTestUtil extends TestCommon {
     )
   }
 
+  def modifyValidatorLicensesWithVoting(
+      svToCreateVoteRequest: SvAppBackendReference,
+      svsToCastVotes: Seq[SvAppBackendReference],
+      changes: Seq[ValidatorLicenseChange],
+      checkClue: String = "validator license modifications have been applied",
+  )(
+      verifyResult: => Unit
+  )(implicit
+      ec: ExecutionContextExecutor
+  ): Unit = {
+    val action = new ARC_DsoRules(
+      new SRARC_ModifyValidatorLicenses(
+        new ValidatorLicensesModification(changes.asJava)
+      )
+    )
+
+    def onlyModifyValidatorLicensesVoteRequests(
+        voteRequests: Seq[Contract[VoteRequest.ContractId, VoteRequest]]
+    ) =
+      voteRequests.filter {
+        _.payload.action match {
+          case arc: ARC_DsoRules =>
+            arc.dsoAction match {
+              case _: SRARC_ModifyValidatorLicenses => true
+              case _ => false
+            }
+          case _ => false
+        }
+      }
+
+    actAndCheck(timeUntilSuccess = 60.seconds)(
+      "Voting on validator license modification", {
+        val (_, voteRequest) = actAndCheck(
```

Comment:
nested actAndCheck is a code smell. Why is it required?

Reply:
I removed the nested here. But from the API caller's perspective there is still a nested `actAndCheck`. I had a look at code elsewhere and found similar nested use of `actAndCheck` (in caller and internally in API), so it should be ok here also.